samples/extensions/open_gl_interop/open_gl_interop.cpp

```diff
@@ -0,0 +1,931 @@
+		gladLoadGL();
+#endif
+
+		glDebugMessageCallback(debugMessageCallback, NULL);
+		glEnable(GL_DEBUG_OUTPUT_SYNCHRONOUS);
+
+		program   = buildProgram(OPENG_VERTEX_SHADER, OPENGL_FRAGMENT_SHADER);
+		startTime = getTime();
+
+		glDisable(GL_DEPTH_TEST);
+
+		// Create the texture for the FBO color attachment.
+		// This only reserves the ID, it doesn't allocate memory
+		glGenTextures(1, &color);
+		glBindTexture(GL_TEXTURE_2D, color);
+
+		// Create the GL identifiers
+
+		// semaphores
+		glGenSemaphoresEXT(1, &glReady);
+		glGenSemaphoresEXT(1, &glComplete);
+		// memory
+		glCreateMemoryObjectsEXT(1, &mem);
+
+		// Platform specific import.
+		glImportSemaphore(glReady, GL_HANDLE_TYPE, handles.glReady);
+		glImportSemaphore(glComplete, GL_HANDLE_TYPE, handles.glComplete);
+		glImportMemory(mem, memorySize, GL_HANDLE_TYPE, handles.memory);
+
+		// Use the imported memory as backing for the OpenGL texture.  The internalFormat, dimensions
+		// and mip count should match the ones used by Vulkan to create the image and determine it's memory
+		// allocation.
+		glTextureStorageMem2DEXT(color, 1, GL_RGBA8, SHARED_TEXTURE_DIMENSION, SHARED_TEXTURE_DIMENSION, mem, 0);
+		glBindTexture(GL_TEXTURE_2D, 0);
+
+		// The remaining initialization code is all standard OpenGL
+		glGenVertexArrays(1, &vao);
+		glBindVertexArray(vao);
+
+		glGenFramebuffers(1, &fbo);
+		glBindFramebuffer(GL_FRAMEBUFFER, fbo);
+		glFramebufferTexture(GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0, color, 0);
+
+		glUseProgram(program);
+		glProgramUniform3f(program, 0, (float) SHARED_TEXTURE_DIMENSION, (float) SHARED_TEXTURE_DIMENSION, 0.0f);
+
+		glViewport(0, 0, SHARED_TEXTURE_DIMENSION, SHARED_TEXTURE_DIMENSION);
+	}
+
+	void render()
+	{
+		float time = (float) (getTime() - startTime);
+		// The GL shader animates the image, so provide the time as input
+		glProgramUniform1f(program, 1, time);
+
+		// Wait (on the GPU side) for the Vulkan semaphore to be signaled
+		GLenum srcLayout = GL_LAYOUT_COLOR_ATTACHMENT_EXT;
+		glWaitSemaphoreEXT(glReady, 0, nullptr, 1, &color, &srcLayout);
+
+		// Draw to the framebuffer
+		glDrawArrays(GL_TRIANGLE_STRIP, 0, 4);
+
+		// Once drawing is complete, signal the Vulkan semaphore indicating
+		// it can continue with it's render
+		GLenum dstLayout = GL_LAYOUT_SHADER_READ_ONLY_EXT;
+		glSignalSemaphoreEXT(glComplete, 0, nullptr, 1, &color, &dstLayout);
+
+		// When using synchronization across multiple GL context, or in this case
+		// across OpenGL and another API, it's critical that an operation on a
+		// synchronization object that will be waited on in another context or API
+		// is flushed to the GL server.
+		//
+		// Failure to flush the operation can cause the GL driver to sit and wait for
+		// sufficient additional commands in the buffer before it flushes automatically
+		// but depending on how the waits and signals are structured, this may never
+		// occur.
+		glFlush();
+	}
+
+	void destroy()
+	{
+		glFinish();
```

Comment:
destroy could be moved to the applications `virtual void finish();` see framework/platform/application.h